drivers/adc/adc_npcx.c

```diff
@@ -719,14 +725,42 @@ static const struct adc_npcx_config adc_npcx_cfg_0 = {
 	.pcfg = PINCTRL_DT_INST_DEV_CONFIG_GET(0),
 };
 
-static struct adc_npcx_threshold_data threshold_data_0;
+static struct adc_npcx_threshold_data threshold_data_0 = {
+	.work_q = &k_sys_work_q,
```

Comment:
Non-blocking comment - you only need to set the default work queue when `CONFIG_ADC_CMP_NPCX_WORKQUEUE=n`.  This will save 4 bytes of flash space when using a separate work queue.